avocado/core/test.py

```diff
@@ -770,6 +789,19 @@ def fetch_asset(self, name, asset_hash=None, algorithm='sha1',
         return asset.Asset(name, asset_hash, algorithm, locations,
                            self.cache_dirs, expire).fetch()
 
+    def recover_failure(self, previous_run):
+        """
+        Method which is called before setUp and other functions to recover if
+        the previous run was interrupted. Avocado does not know about the tests
+        state, for recovery only the incomplete result object is provided
+        which may allow recovering some data from the tests log directory.
+
+        :param previous_run: Stored test status from previous run
+        """
+
```

Comment:
Where is the recovery code? Anyway, this could be simply conceptual. A test that was interrupted during it's execution should be, imo, executed again in the `resume` command.